app/controllers/users_controller.rb

```diff
@@ -0,0 +1,67 @@
+class UsersController < ApplicationController
+  # before_action :authenticate_user, except: %i[create signout]
+  before_action :set_user, only: %i[show update destroy]
+
+  # GET /users
+  def index
+    @users = User.all
+    render json: @users
+  end
+
+  # GET /users/1
+  def show
+    render json: @user
+  end
+
+  def logged_in?
+    if current_user
+      render json: { current_user: current_user.username }, status: :ok
+    else
+      render json: { current_user: nil }, status: 401
+    end
+  end
+
+  # POST /users
+  def create
+    @user = User.new(user_params)
+
+    if @user.save
+      auth_token = Knock::AuthToken.new(payload: { sub: @user.id })
+      session[:jwt] = auth_token.token
+      render json: @user, status: :created
+    else
+      render json: @user.errors.full_messages, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /users/1
+  def update
+    if @user.update(user_params)
+      render json: @user
+    else
+      render json: @user.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /users/1
+  def destroy
+    @user.destroy
+  end
+
+  def signout
```

Comment:
Again, the back end/RESTful API isn't responsible for keeping authenticated state (it's Stateless) and this controller action name isn't RESTful at all

Reply:
Sign out method has been refactored to the destroy method. The authentication used is stateless as there is no session data being tracked in rails. Here, we are using session to read the cookie sent from the user and we are removing the token in it. The cookie is http only for security and hence cannot be removed on the frontend. This is why we have a http route to clear cookies. This is deterministic and stateless.

Reply:
I get what you mean. Though, since you're trying to delete JWT from sessions here you are effectively trying to keep authentication state here in the back end - which as previously mentioned isn't what your API should be doing since it's supposed to be stateless.

the session object doesn't really do anything here and in fact isn't being used at all so it's just an object that floats.

Reply:
I researched and found that the best way to store jwt token on the frontend is to put it in http only cookies. I tried deleting the cookie with jwt on the frontend on signout but failed as this is a http only cookie and can only be cleared on the backend  which is what I am doing here. Please suggest me how I can solve this issue keeping it stateless and secure as well

Reply:
It might take me very long to explain what i mean here. Just refer to this short article on Microverse questions that I wrote -> https://questions.microverse.org/t/how-to-properly-implement-authentication-when-using-rails-to-create-apis/1818?u=rvvergara